mne/utils/_bunch.py

```diff
@@ -94,10 +96,58 @@ def _named_subclass(klass):
 class NamedInt(_Named, int):
     """Int with a name in __repr__."""
 
-    pass
-
 
 class NamedFloat(_Named, float):
     """Float with a name in __repr__."""
 
-    pass
+
+class MNEObjectsList(MutableSequence):
+    """All the bolierplate for a list of specific MNE objects.
+
+    Parameters
+    ----------
+    elements : list
+        A list of Objects objects.
+
+    Attributes
+    ----------
+    _items : list
+        The container
+    """
+
+    def __init__(self, elements=None, kls=None):
+        if kls is None:
+            raise ValueError('kls is necessary')
+        if elements is None:
+            self._items = list()
+        elif all([isinstance(_, kls) for _ in elements]):
+            if elements is None:
+                self._items = list()
```

Comment:
move this on the previous condition.
`if elements is None or (isinstance(elements, list) and not elements):`